### ISSUE TITLE
Return a diskRef that can delete the entire gopath root

### ISSUE THREAD

**Issue by arschles** [gomods/athens#329]
After #291, the disk ref (in `./pkg/module/disk_ref.go`) will return a `diskRef` that will clean up the zip, info and mod files. It should delete the _entire_ GOPATH root directory that it owns

**Comment by arschles** [comment]
Also, when this is done, the `TestGoGetFetcherFetch()` test in `./pkg/module/go_get_fetcher_test.go` should be updated to test that the entire GOPATH is deleted for that ref

**Comment by robjloranger** [comment]
is this a duplicate of #318?
